### PR TITLE
GMU-3 - fixed ClassCastException

### PR DESCRIPTION
**Pull request by Igor-Gorchakov**
**Purpose**
As it turned out the JsonPathSyntaxReader doesn't support paths to read an array of SimpleFields from embedded objects, for example `$.holdings[*].hrid` , `$.holdings[*].items[*].yearCaption` 

**Approach**
Using the same approach to build SimpleValue the same as it works for CompositeValue.
Changed the ListValue to be parametrized `List<SimpleValue>`, not `List<String>` as before
Step1 - read matrix, take only 1st entry because of SimpleValue
Step2 - build SimpleValue from the entry. Return StringValue of ListValue depending on the given entry

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,206 +1,206 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
-  <modelVersion>4.0.0</modelVersion>
```
Comment (andrii-novytskyi) [inline pom.xml:2]: what`s wrong with pom? why all file new?

File: src/main/java/org/folio/reader/JPathSyntaxEntityReader.java [hunk 2]
```diff
@@ -45,6 +46,59 @@ public JPathSyntaxEntityReader(JsonObject entity) {
...
+      }
+    }
+  }
+
+  /**
+   * Builds a SimpleValue, could be StringValue or ListValue
+   */
+  private SimpleValue buildSimpleValue(DataSource dataSource, List<ValueWrapper> valueWrappers, Object nonNullValue) {
+    boolean isStringRuleValue = false;
+    boolean isListRuleValue = false;
```
Comment (illia-borysenko) [inline src/main/java/org/folio/reader/JPathSyntaxEntityReader.java:71]: I don't think that usage of these two fields is necessary,  you can just use the condition bellow in the if/else statement

**Comment by KVupp** [comment]
Is the description of the issue correct? Because that's not what I see in the evaluator:
Screen Shot 2020-12-16 at 9 06 32 AM

**Comment by Igor-Gorchakov** [comment]
> Is the description of the issue correct? Because that's not what I see in the evaluator:
> Screen Shot 2020-12-16 at 9 06 32 AM

Hi @KVupp, the reader expected to work only with **string value** using such path `$.holdings[*].hrid`, but JsonPath returns **array value** that causes ClassCastException

**Review by andrii-novytskyi — APPROVED** [review]

**Review by illia-borysenko — APPROVED** [review]

**Pull request merged**